test/jdk/java/net/httpclient/http2/IdleConnectionTimeoutTest.java

```diff
@@ -0,0 +1,161 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8263031
+ * @summary Tests that the HttpClient can correctly receive a Push Promise
+ *          Frame with the END_HEADERS flag unset followed by one or more
+ *          Continuation Frames.
+ * @library /test/lib server
+ * @build jdk.test.lib.net.SimpleSSLContext
+ * @modules java.base/sun.net.www.http
+ *          java.net.http/jdk.internal.net.http.common
+ *          java.net.http/jdk.internal.net.http.frame
+ *          java.net.http/jdk.internal.net.http.hpack
+ * @run testng/othervm IdleConnectionTimeoutTest
+ */
+
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CompletableFuture;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.net.http.HttpClient.Version.HTTP_2;
+import static org.testng.Assert.assertEquals;
+
+public class IdleConnectionTimeoutTest {
+
+    Http2TestServer http2TestServer;
+    URI timeoutUri;
+    URI noTimeoutUri;
+    final String IDLE_CONN_PROPERTY = "jdk.httpclient.idleConnectionTimeout";
+    final String TIMEOUT_PATH = "/serverTimeoutHandler";
+    final String NO_TIMEOUT_PATH = "/noServerTimeoutHandler";
+    static boolean expectTimeout;
+    static final PrintStream testLog = System.err;
+
+    @BeforeTest
+    public void setup() throws Exception {
+        http2TestServer = new Http2TestServer(false, 0);
+        http2TestServer.addHandler(new ServerTimeoutHandler(), TIMEOUT_PATH);
+        http2TestServer.addHandler(new ServerNoTimeoutHandler(), NO_TIMEOUT_PATH);
+
+        http2TestServer.start();
+        int port = http2TestServer.getAddress().getPort();
+        timeoutUri = new URI("http://localhost:" + port + TIMEOUT_PATH);
+        noTimeoutUri = new URI("http://localhost:" + port + NO_TIMEOUT_PATH);
+    }
+
+    /*
+       If the InetSocketAddress of the first remote connection is not equal to the address of the
+       second remote connection, then the idleConnectionTimeoutEvent has occurred and a new connection
+       was made to carry out the second request by the client. Otherwise, the old connection was reused.
+    */
+    @Test
+    public void testTimeoutFires() throws InterruptedException {
+        expectTimeout = true;
+        System.setProperty(IDLE_CONN_PROPERTY, "100");
```

Comment:
Once we change our implementation to read the system property just once and cache it, we will then have to move setting this system property as a `-Djdk.httpclient.idleConnectionTimeout` value in the test definition to make sure it does indeed get picked up correctly.